provider/pkg/provider/provider.go

```diff
@@ -2472,7 +2473,8 @@ func (k *kubeProvider) Delete(ctx context.Context, req *pulumirpc.DeleteRequest)
 			Resources:         resources,
 			ServerSideApply:   k.serverSideApplyMode,
 		},
-		Inputs:  current,
+		Inputs:  oldInputs,
+		Outputs: current,
```

Comment:
Important to note here that the `Inputs` field here contained the _old outputs_ (`current`). The deletion logic should therefore use `Outputs` to maintain the status-quo.

A subtle (positive?) consequence is that the `timeout` annotation is drawn from the live object.